modules/sdk/project-templates/project-templates-rest-builder/src/test/java/com/liferay/project/templates/rest/builder/ProjectTemplatesRESTBuilderTest.java

```diff
@@ -118,12 +116,234 @@ public void testBuildTemplateRESTBuilderCheckExports() throws Exception {
 		Assert.assertArrayEquals(
 			Files.readAllBytes(gradleOpenAPIYAML.toPath()),
 			Files.readAllBytes(mavenOpenAPIYAML.toPath()));
+	}
+
+	@Test
+	public void testBuildTemplateRESTBuilderCheckExports() throws Exception {
+		String liferayVersion = getDefaultLiferayVersion();
+		String name = "guestbook";
+		String packageName = "com.liferay.docs.guestbook";
+		String template = "rest-builder";
+
+		File gradleWorkspaceDir = buildWorkspace(
+			temporaryFolder, "gradle", "gradleWS", liferayVersion,
+			mavenExecutor);
+
+		File gradleWorkspaceModulesDir = new File(
+			gradleWorkspaceDir, "modules");
+
+		File gradleProjectDir = buildTemplateWithGradle(
+			gradleWorkspaceModulesDir, template, name, "--package-name",
+			packageName, "--liferay-version", liferayVersion);
+
+		File gradleBndFile = new File(
+			new File(gradleProjectDir, name + "-api"), "bnd.bnd");
+
+		Assert.assertTrue(gradleBndFile.exists());
+
+		File mavenWorkspaceDir = buildWorkspace(
+			temporaryFolder, "maven", "mavenWS", liferayVersion, mavenExecutor);
+
+		File mavenModulesDir = new File(mavenWorkspaceDir, "modules");
+
+		File mavenProjectDir = buildTemplateWithMaven(
+			mavenModulesDir, mavenModulesDir, template, name, "com.test",
+			mavenExecutor, "-Dpackage=" + packageName,
+			"-DliferayVersion=" + liferayVersion);
+
+		File mavenBndFile = new File(
+			new File(mavenProjectDir, name + "-api"), "bnd.bnd");
+
+		Assert.assertTrue(mavenBndFile.exists());
+
+		Assert.assertArrayEquals(
+			Files.readAllBytes(gradleBndFile.toPath()),
+			Files.readAllBytes(mavenBndFile.toPath()));
 
 		testContains(
 			gradleProjectDir, name + "-api/bnd.bnd", "Export-Package:\\",
 			packageName + ".dto.v1_0,\\", packageName + ".resource.v1_0");
 	}
 
+	@Test
+	public void testBuildTemplateRESTBuilderCheckIncludeResources()
+		throws Exception {
+
+		String extraModules = "true";
+		String liferayVersion = getDefaultLiferayVersion();
+		String name = "guestbook";
+		String template = "rest-builder";
+
+		File gradleWorkspaceDir = buildWorkspace(
+			temporaryFolder, "gradle", "gradleWS", liferayVersion,
+			mavenExecutor);
+
+		File gradleWorkspaceModulesDir = new File(
+			gradleWorkspaceDir, "modules");
+
+		File gradleProjectDir = buildTemplateWithGradle(
+			gradleWorkspaceModulesDir, template, name, "--liferay-version",
+			liferayVersion, "--extraModules", extraModules);
+
+		File gradleTestBndFile = new File(
+			new File(gradleProjectDir, name + "-test"), "bnd.bnd");
+
+		Assert.assertTrue(gradleTestBndFile.exists());
+
+		testContains(
```

Comment:
I think we should add this check as a part of `testBuildTemplateRESTBuilderWithExtraModules()`. Since this is the only check of this test and we need to generate the *-test module there anyway

Reply:
I'll add it